packages/squiggle-lang/src/dist/BaseDist.ts

```diff
@@ -34,6 +34,11 @@ export abstract class BaseDist {
     return Result.fmap(this.variance(), Math.sqrt);
   }
   abstract variance(): result<number, DistError>;
+  protected abstract _isEqual(b: BaseDist): boolean;
```

Comment:
This part seems kind of awkward, wasn't really sure how to do it. Basically, each class is responsible for a ``_isEqual()`` function that just works for elements of that class, but the BaseDist puts those together with its own ``isEqual()`` function, that works for any BaseDists.